be/src/runtime/runtime_state.cpp

```diff
@@ -425,6 +454,17 @@ Status RuntimeState::append_error_msg_to_file(std::function<std::string()> line,
     return Status::OK();
 }
 
+std::string RuntimeState::get_error_log_file_path() const {
+    if (config::save_load_error_log_to_s3 && config::is_cloud_mode()) {
+        // expiration must be less than a week (in seconds) for presigned url
+        static const unsigned EXPIRATION_SECONDS = 7 * 24 * 60 * 60 - 1;
+        // We should return a public endpoint to user.
+        return std::dynamic_pointer_cast<io::S3FileSystem>(_s3_error_fs)
```

Comment:
We need to check every cast `std::dynamic_pointer_cast<io::S3FileSystem>(_s3_error_fs)` 
because it may be nullptr if the storage backend is HDFS only.